Documentation/ApiOverview/TSFE/Index.rst

```diff
@@ -0,0 +1,234 @@
+    }
+
+.. _tsfe_pageId:
+
+Access current page ID
+----------------------
+
+Access the current page ID:
+
+.. code-block:: php
+
+    // this is discouraged
+    $pageId = $GLOBALS['TSFE']->id;
+
+Can be done using the :ref:`'routing' <typo3-request-attribute-routing>`
+request attribute:
+
+.. code-block:: php
+
+    $pageArguments = $request->getAttribute('routing');
+    $pageId = $pageArguments->getPageId();
+
+.. _tsfe_language:
+
+Access language settings
+------------------------
+
+In order to get current language settings, such as the current language ID,
+obtain :php:`\TYPO3\CMS\Core\Site\Entity\SiteLanguage` object from the
+:ref:`request attribute <request-attributes>` 'language':
+
+.. code-block:: php
+
+    // outdated TSFE method, use site settings instead
+    $languageId = $GLOBALS['TSFE']->sys_language_uid;
+
+.. code-block:: php
+
+    // TYPO3\CMS\Core\Site\Entity\SiteLanguage object.
+    $language = $request->getAttribute('language');
+    $languageId = $language->getLanguageId();
+
+
+If the request is not available, accessing language settings
+can be done using the :ref:`language aspect <context_api_aspects_language>`.
+
+Get the language of the current page as integer:
+
+.. code-block:: php
+
+    $languageId = (int) $context->getPropertyFromAspect('language', 'id');
+
+.. _tsfe_frontendUser:
+
+Access frontend user information
+--------------------------------
+
+Accessing information about frontend users. Accessing
+:php:`$GLOBALS['TSFE']->fe_user` directly is discouraged.
+
+.. code-block:: php
+
+    // discouraged, use alternatives if possible
+    // TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication
+    $feUser = $GLOBALS['TSFE']->fe_user;
+
+Using the request attribute :ref:`frontend.user <typo3-request-attribute-frontend-user>`:
+
+.. code-block:: php
+
+    // TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication
+    $frontendUser = $request->getAttribute('frontend.user');
+
+Some information via frontend and backend users con be obtained via the
+:ref:`user aspect <context_api_aspects_user>`. For example:
+
+.. code-block:: php
+
+    // return whether a frontend user is logged in
+    $context->getPropertyFromAspect('frontend.user', 'isLoggedIn');
+
+.. _tsfe_baseURL:
+
+Get current base URL
+--------------------
+
+To get the base URL of the current site (or other :ref:`site configuration <sitehandling>`), use site
+configuration:
+
+.. code-block:: php
+
+    // It used to be possible to get the baseUrl configuration (from TypoScript
+    // config.baseURL) this way.
+    // The property is now protected and deprecated since TYPO3 v12,
+    // use site configuration
+    $GLOBALS['TSFE']->baseURL
+
+.. code-block:: php
+
+    // TYPO3\CMS\Core\Site\Entity\Site
+    $site = $request->getAttribute('site');
+    // array
+    $siteConfiguration = $site->getConfiguration();
+    $baseUrl = $siteConfiguration['base'];
+
+.. _tsfe_siteByPageId:
+
+Get site by page ID
```

Comment:
I am missing the relation of site to TSFE. This is an example how to get the site. Is/was it possible with TSFE to get the site?

Reply:
yes `TSFE->id` or `$GLOBALS['TSFE']->id`

Reply:
Hmm, this is the page ID, not getting the site through the page ID?

The section should point to it: TSFE (discouraged), SiteFinder (the way to go).

Reply:
oh sorry.
I don't know how to get an arbitrary site by page ID.  
Just the current page was something like  `$GLOBALS['TSFE']->page`, don't remember exactly

Reply:
Ah, there is a public TSFE->getSite() method. This should be added as discouraged example to show the relation how to do better.

Reply:
I think, we should remove the section. If we leave it, also getSiteLanguage(), getContext(), etc. should be documented. But there were introduced with v9 AFAIK, so there were better APIs in place already.

Reply:
I removed this section. It is true. It is not directly related. We already have a page id and a language id example.